src/Resource/MoipResource.php

```diff
@@ -134,24 +134,71 @@ public function jsonSerialize()
     }
 
     /**
-     * Find by path.
+     * Execute a http request. If payload == null no body will be sent. Empty body ('{}') is supported by sending a
+     * empty stdClass.
      *
-     * @param string $path
+     * @param            $path
+     * @param            $method
+     * @param mixed|null $payload
+     *
+     * @throws Exceptions\ValidationException  if the API returns a 4xx http status code. Usually means invalid data was sent.
+     * @throws Exceptions\UnautorizedException if the API returns a 401 http status code. Check API token and key
+     * @throws Exceptions\UnexpectedException  if the API returns a 500 http status code, this is not suppose to happen. Please report the error to moip
      *
      * @return stdClass
      */
-    public function getByPath($path)
+    protected function httpRequest($path, $method, $payload = null)
     {
         $httpConnection = $this->createConnection();
         $httpConnection->addHeader('Content-Type', 'application/json');
+        if ($payload !== null) {
+            // if it's json serializable
+            $body = json_encode($payload, JSON_UNESCAPED_SLASHES);
+            if ($body) {
+                $httpConnection->addHeader('Content-Length', strlen($body));
+                $httpConnection->setRequestBody($body);
+            }
+        }
 
-        $httpResponse = $httpConnection->execute($path, HTTPRequest::GET);
-
-        if ($httpResponse->getStatusCode() != 200) {
-            throw new RuntimeException($httpResponse->getStatusMessage(), $httpResponse->getStatusCode());
+        /*
+         * @var \Moip\Http\HTTPResponse
+         */
+        $http_response = $httpConnection->execute($path, $method);
+
+        $code = $http_response->getStatusCode();
+
+        if ($code >= 200 && $code < 299) {
+            return json_decode($http_response->getContent());
+        } elseif ($code == 401) {
+            throw new Exceptions\UnautorizedException();
+        } elseif ($code >= 400 && $code <= 499) {
+            $error_obj = json_decode($http_response->getContent());
+            $errors = [];
+            if ($error_obj && isset($error_obj->errors)) { // just in case
+                foreach ($error_obj->errors as $error) {
+                    $errors[] = new Exceptions\Error($error->code, $error->path, $error->description);
+                }
+                throw new Exceptions\ValidationException($code, $http_response->getStatusMessage(), $errors);
+            } else {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method httpRequest uses an else expression. Else is never necessary and you can simplify the code to work without else.](https://www.codacy.com/app/jeancesargarcia/moip-sdk-php/file/2076273159/issues/source?bid=2821642&fileBranchId=3138004#l182)